src/main/scala/io/github/evis/scalafix/maven/plugin/mojo/ScalafixMojo.scala

```diff
@@ -84,19 +60,59 @@ final class ScalafixMojo extends AbstractMojo {
       getLog.info(
         "Skip scalafix since both skip.main and skip.test flags passed")
     } else {
+      val build = project.getBuild
+      val buildPath = getPath(build.getDirectory)
+      val mainSources = getSources(
+        skipMain,
+        "main",
+        buildPath,
+        mainSourceDirectories,
+        project.getCompileSourceRoots,
+        build.getSourceDirectory
+      )
+      val testSources = getSources(
+        skipTest,
+        "test",
+        buildPath,
+        testSourceDirectories,
+        project.getTestCompileSourceRoots,
+        build.getTestSourceDirectory
+      )
+      val mainOutputs =
+        if (skipMain) Seq.empty else Seq(build.getOutputDirectory)
+      val testOutputs =
+        if (skipTest) Seq.empty else Seq(build.getTestOutputDirectory)
       val params =
         List(
-          SourceDirectoryParam(sourceDirectory).ifNot(skipMain),
-          SourceDirectoryParam(testSourceDirectory).ifNot(skipTest),
-          ProjectDependenciesParam(projectDependencies),
-          CompiledDirectoryParam(compiledDirectory),
-          CompiledDirectoryParam(testCompiledDirectory),
-          PluginsParam(plugins),
+          SourceDirectoryParam(mainSources ++ testSources),
+          ProjectDependenciesParam(project.getArtifacts.asScala),
+          CompiledDirectoryParam(mainOutputs ++ testOutputs),
+          PluginsParam(build.getPlugins.asScala),
           ModeParam(mode),
           ConfigParam(config),
           CommandLineArgsParam(commandLineArgs)
         )
       run(params, getLog)
     }
   }
+
+  private def getSources(
+      flag: Boolean,
+      flagName: String,
+      outputDir: Path,
+      customDirs: JList[File],
+      projectDirs: JList[String],
+      projectDir: String): Iterable[File] = {
+    val dirs =
+      if (flag) Seq.empty
+      else if (!customDirs.isEmpty) customDirs.asScala
+      else
+        new File(projectDir, "/../scala") +: projectDirs.asScala.flatMap { x =>
+          val file = getFile(x)
+          if (file.toPath.startsWith(outputDir)) None else Some(file)
+        }
+    if (dirs.isEmpty) getLog.info(s"Skip scalafix[$flagName]")
+    else dirs.foreach(dir => getLog.info(s"Processing[$flagName]: $dir"))
```

Comment:
Now this code prints:

```
[INFO] Processing[test]: <project_dir>/src/test/scala/../scala
```

`scala/../scala` looks confusing. I suggest to make it canonical before logging:

```suggestion
    else dirs.foreach(dir => getLog.info(s"Processing[$flagName]: ${dir.getCanonicalPath}"))
```